cpp/daal/src/algorithms/dtrees/forest/df_train_dense_default_impl.i

```diff
@@ -384,12 +491,16 @@ protected:
             _minWeightLeaf       = par.minWeightFractionInLeafNode * x->getNumberOfRows(); // no sample_weight
             _minImpurityDecrease = par.minImpurityDecreaseInSplitNode * x->getNumberOfRows()
                                    - daal::services::internal::EpsilonVal<algorithmFPType>::get() * x->getNumberOfRows();
+            _remainingSplitNodes = par.maxLeafNodes;
         }
     }
 
     size_t nFeatures() const { return _data->getNumberOfColumns(); }
     typename DataHelper::NodeType::Base * build(services::Status & s, size_t iStart, size_t n, size_t level,
                                                 typename DataHelper::ImpurityData & curImpurity, bool & bUnorderedFeaturesUsed, size_t nClasses);
+    typename DataHelper::NodeType::Base * buildBF(services::Status & s, size_t iStart, size_t n, size_t level,
```

Comment:
I suggest to avoid BF term here, because of ambiguity: best-first vs breadth-first vs brute-force. buildBestFirst name works here?

Reply:
it was done